File: tests/check_versions.py
#!/usr/bin/env python3
"""CI check: roar_sdk.__spec_version__ must match spec/VERSION.json spec_version."""

import json
import sys
from pathlib import Path

try:
    import roar_sdk
except ImportError:
    print("FAIL: roar_sdk not installed. Run: pip install -e ./python")
    sys.exit(1)

spec = json.loads((Path(__file__).parent.parent / "spec" / "VERSION.json").read_text())
sdk_ver = roar_sdk.__spec_version__
spec_ver = spec["spec_version"]

if sdk_ver != spec_ver:
    print(f"FAIL: roar_sdk.__spec_version__ ({sdk_ver!r}) != spec/VERSION.json ({spec_ver!r})")
    print("Update roar_sdk/__init__.py __spec_version__ to match the spec version.")
    sys.exit(1)

print(f"Version aligned: {sdk_ver!r}")

File: tests/check_versions.mjs
#!/usr/bin/env node
/**
 * CI check: @roar-protocol/sdk version in package.json must be coherent,
 * and ts_sdk_min_version in spec/VERSION.json must be set.
 *
 * Mirrors tests/check_versions.py logic for the TypeScript SDK.
 */

import { readFileSync } from "fs";
import { resolve, dirname } from "path";
import { fileURLToPath } from "url";

const __dirname = dirname(fileURLToPath(import.meta.url));
const root = resolve(__dirname, "..");

const spec = JSON.parse(readFileSync(resolve(root, "spec", "VERSION.json"), "utf8"));
const pkg = JSON.parse(readFileSync(resolve(root, "ts", "package.json"), "utf8"));

// 1. Package name must be correct
if (pkg.name !== "@roar-protocol/sdk") {
  console.error(`FAIL: ts/package.json name is ${JSON.stringify(pkg.name)}, expected "@roar-protocol/sdk"`);
  process.exit(1);
}

// 2. SDK version must be semver (basic check)
if (!/^\d+\.\d+\.\d+/.test(pkg.version)) {
  console.error(`FAIL: ts/package.json version ${JSON.stringify(pkg.version)} is not semver`);
  process.exit(1);
}

// 3. ts_sdk_min_version must be set in VERSION.json (was null in initial release)
if (!spec.compatibility?.ts_sdk_min_version) {
  console.warn(`WARN: spec/VERSION.json compatibility.ts_sdk_min_version is not set — update it to ${pkg.version}`);
  // Not a hard failure — will be enforced once we bump the spec
}

// 4. Python and TS SDK major.minor must match (they track together)
const pyMinVer = spec.compatibility?.python_sdk_min_version ?? "";
const tsSdkVer = pkg.version;
const [pyMaj, pyMin] = pyMinVer.split(".").map(Number);
const [tsMaj, tsMin] = tsSdkVer.split(".").map(Number);
if (pyMaj !== tsMaj || pyMin !== tsMin) {
  console.error(
    `FAIL: Python SDK min version (${pyMinVer}) and TS SDK version (${tsSdkVer}) ` +
    `have different major.minor — they should track together.`
  );
  process.exit(1);
}

console.log(`TS SDK version aligned: ${pkg.version} (spec ${spec.spec_version})`);
